typed/remove.go

```diff
@@ -92,12 +104,24 @@ func (w *removingWalker) doList(t *schema.List) (errs ValidationErrors) {
 }
 
 func (w *removingWalker) doMap(t *schema.Map) ValidationErrors {
+	if !w.value.IsMap() {
```

Comment:
I'm curious what use-case this covers? I'm guessing if we have for example. In that case both remove or extract can't do anything beside return nil . Is that right?

Reply:
Ya, the use case we see is any time an extract path matches to a a top-level struct that maps to a nil value.

In t[his example](https://github.com/kubernetes-sigs/structured-merge-diff/pull/188/files#diff-7f11c34fef592894fdf658dda9b43b5ee8f036a06620052c41b5144805cfd9f2R692-R699), we have a match on path `.c` so in `m.Iterate` we hit the [new recursion](https://github.com/kubernetes-sigs/structured-merge-diff/pull/188/files#diff-9bfd5d4edd67f21a522024f4f5725b30dffdceada9af3fee0310ab699755401cR145) we added to call `removeItemsWithSchema()` for the key `.c` and the val `&{Value:<nil>}` 

When we recurse into `doMap` now [the line](https://github.com/kubernetes-sigs/structured-merge-diff/pull/188/files#diff-9bfd5d4edd67f21a522024f4f5725b30dffdceada9af3fee0310ab699755401cR110) `m := w.value.AsMapUsing(w.allocator)` will panic if we don't do the isMap check first.